src/modules/extrakeys/main_impl.h

```diff
@@ -55,6 +55,19 @@ int secp256k1_xonly_pubkey_serialize(const secp256k1_context* ctx, unsigned char
     return 1;
 }
 
+int secp256k1_xonly_pubkey_cmp(const secp256k1_context* ctx, const secp256k1_xonly_pubkey* pk1, const secp256k1_xonly_pubkey* pk2) {
+    unsigned char out1[32];
+    unsigned char out2[32];
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(pk1 != NULL);
+    ARG_CHECK(pk2 != NULL);
+
+    CHECK(secp256k1_xonly_pubkey_serialize(ctx, out1, pk1));
```

Comment:
Not sure why these are `CHECK` instead of `VERIFY_CHECK`.

Reply:
`CHECK` and `VERIFY_CHECK` are both not right here because they abort if the check fails, making it impossible to test the function with illegal keys. I'd suggest to remove the `CHECK` and instead make sure that `out1` is all 0 if `pubkey_serialize` fails.